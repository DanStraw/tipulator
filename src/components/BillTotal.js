import React from 'react';
import { View, Text } from 'react-native';
import { connect } from 'react-redux';

const mapState = state => {
  const { totalBill } = state;
  return { totalBill };
}

const BillTotal = props => {
  const { totalBill } = props;
  if (!totalBill) {
    return (
      <View>
        <Text testID="total-bill" total-bill="0.00">$0.00</Text>
      </View>
    )
  }

  return (
    <View>
      <Text>$</Text><Text testID="total-bill">{totalBill}</Text>
    </View>
  )
}

export default connect(mapState)(BillTotal);