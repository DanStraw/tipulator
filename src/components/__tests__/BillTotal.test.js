import React from 'react';
import { Provider } from 'react-redux';
import store from '../../store';
import BillTotal from '../BillTotal';
import { render } from '@testing-library/react-native';


describe('<BillTotal /> component ', () => {
  test('exists and text value defaults to $0.00', async () => {
    const component = (
      <Provider store={store}>
        <BillTotal />
      </Provider>
    );
    const { findByText } = render(component);
    const newBillTotal = await findByText('$0.00');
    expect(newBillTotal).toBeTruthy();
  });

  test('get component by testID', () => {

    const component = (
      <Provider store={store}>
        <BillTotal totalBill={45} />
      </Provider>
    )

    const { getAllByTestId } = render(component);
    const newBillTotal = getAllByTestId('total-bill');

    expect(newBillTotal.length).toBe(1);
  })
})