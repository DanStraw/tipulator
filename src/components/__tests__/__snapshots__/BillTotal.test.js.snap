// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<BillTotal /> component  exists and text value defaults to $0.00 1`] = `
<Text
  testID="total-bill"
  total-bill="0.00"
>
  $0.00
</Text>
`;

exports[`<BillTotal /> component  get component by testID 1`] = `
<Text
  testID="total-bill"
  total-bill="0.00"
>
  $0.00
</Text>
`;

exports[`<BillTotal /> component no prop <BillTotal /> component with prop 1`] = `
<Text
  testID="total-bill"
  total-bill="0.00"
>
  $0.00
</Text>
`;

exports[`<BillTotal /> component no prop exists and text value defaults to $0.00 1`] = `
<Text
  testID="total-bill"
  total-bill="0.00"
>
  $0.00
</Text>
`;
